components/gitpod-protocol/go/gitpod-service.go

```diff
@@ -57,7 +57,7 @@ type APIInterface interface {
 	SendHeartBeat(ctx context.Context, options *SendHeartBeatOptions) (err error)
 	WatchWorkspaceImageBuildLogs(ctx context.Context, workspaceID string) (err error)
 	IsPrebuildDone(ctx context.Context, pwsid string) (res bool, err error)
-	SetWorkspaceTimeout(ctx context.Context, workspaceID string, duration *WorkspaceTimeoutDuration) (res *SetWorkspaceTimeoutResult, err error)
+	SetWorkspaceTimeout(ctx context.Context, workspaceID string, duration string) (res *SetWorkspaceTimeoutResult, err error)
```

Comment:
```suggestion
	SetWorkspaceTimeout(ctx context.Context, workspaceID string, duration time.Duration) (res *SetWorkspaceTimeoutResult, err error)
```

Reply:
It is not time.Duration but a string with the format `<number>m|h|d`

Reply:
Sure, but the argument to this Interface function can be a `time.Duration`. Inside the method, you can do `duration.String()` which gives you the string representation. The data types on the Go interface should be native, irrespective of what you send over the wire

Reply:
It would break the existing clients.

Reply:
I'm not sure I follow, how would it break them?

The implementation would be as follows:
```golang
func (gp *APIoverJSONRPC) SetWorkspaceTimeout(ctx context.Context, workspaceID string, duration time.Duration) (res *SetWorkspaceTimeoutResult, err error) {
	durationAsString := duration.String()

	if gp == nil {
		err = errNotConnected
		return
	}
	var _params []interface{}
	_params = append(_params, workspaceID)
	_params = append(_params, durationAsString)
	var result SetWorkspaceTimeoutResult
	err = gp.C.Call(ctx, "setWorkspaceTimeout", _params, &result)
	if err != nil {
		return
	}
	res = &result
	return
}
```

Reply:
Existing clients pass in a string

Reply:
But those clients are using the old version of this client library, right? So they will continue to work just as well. New clients (with updated version) would pass 180m.

The request made over the protocol does pass in "180m" due to the `duration.String()` conversion.

This interface is not tied to the underlying websocket protocol.

Reply:
Ah, right it is client-side. In that case, yes, I can do that change